lib/cacheWrapper.js

```diff
@@ -0,0 +1,155 @@
+/* jslint node: true */
+'use strict';
+
+var _ = require( 'lodash' );
+var Catbox = require( 'catbox' );
+var Memory = require( 'catbox-memory' );
+var Redis = require( 'catbox-redis' );
+var Q = require( 'q' );
+
+var cacheClient;
+var startingClient = false;
+var queryQueue = [];
+var policies = [];
+
+function set( options ) {
+	options.method = 'stash';
+	return addCallToQueue( options );
+}
+
+function get( options ) {
+	options.method = 'retrieve';
+	return addCallToQueue( options );
+}
+
+/**
+* Single interface to the cache plugin so we can manage the cacheClient effectively
+* @param {Object} options - Including segment, key and value.
+*/
+function addCallToQueue( options ) {
+	var deferred = Q.defer();
+	if( !cacheClient ) {
+		deferred.reject( new Error( 'Cache not intialised' ) );
+	} else {
+		if( !cacheClient.isReady() ) {
+			// cache isnt ready, queue it up
+			queueRequest( deferred, options );
```

Comment:
(linked to earlier comment) 

I think if you didn't pass in the deferred to these functions you could return early and just write
if( !cacheClient.isReady() ) {
  return queueRequest( options );
}

// execute the requested method straight away
if( options.method === 'stash' ) {
 return stash( options );
}

return retrieve( options );